core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -442,9 +444,29 @@ private long getCurrentBlockId() {
    */
   private long getBlockId(long pos) {
     int index = (int) (pos / mBlockSize);
-    Preconditions.checkState(index < mStatus.getBlockIds().size(),
-        PreconditionMessage.ERR_BLOCK_INDEX.toString(), index, pos, mStatus.getBlockIds().size());
-    return mStatus.getBlockIds().get(index);
+    return mBlockIds.get(index);
+  }
+
+  /**
+   * @param pos the position, assumed to be in the range from 0 to (file length - 1), otherwise, the
+   *        returned block index will be out of range
+   * @return the block index corresponding to the position
+   */
+  private int getBlockIndex(long pos) {
+    return (int) (pos / mBlockSize);
+  }
+
+  /**
+   * @param pos the position, assumed to be in the range from 0 to (file length - 1)
+   * @return whether the position is in current block, if the current block stream is not created
+   *         yet, return false
+   */
+  private boolean isInCurrentBlock(long pos) {
+    // Instead of computing the block index with (pos / mBlockSize) and compare against mBlockIndex,
+    // multiplication is used, this optimization is based on the fact that multiplication is much
+    // faster than division.
+    return mBlockIndex != UNINITIALIZED_BLOCK_INDEX && pos >= mBlockIndex * mBlockSize
```

Comment:
Are there any issue with overflow?

Also, would saving `mBlockIndex * mBlockSize` and then adding `mBlockSize` improve this, or will the compiler take care of that?

Reply:
used two new fields  `mCurrentBlockStartPos` and `mCurrentBlockEndPos` to save multiplication.